### PR TITLE
Buidler build plugin support

### PR DESCRIPTION
**Pull request by Malcador**

### CODE REVIEW

File: commands/export.go [hunk 1]
```diff
@@ -384,7 +384,7 @@ func contractsWithConfig(
 		return nil, nil, err
 	}
 
-	contracts, _, err := providers.GetContracts(providerConfig.AbsoluteBuildDirectoryPath(), []string{networkId}, objects...)
+	contracts, _, err := deploymentProvider.GetContracts(providerConfig.AbsoluteBuildDirectoryPath(), []string{networkId}, objects...)
 
 	var configPayload *payloads.Config
 	if providerConfig.ConfigType == truffle.NewTruffleConfigFile && providerConfig.Compilers != nil {
```
Comment (nebojsa94) [inline commands/export.go:390]: Add handling for BuidlerConfigFile

File: commands/push.go [hunk 2]
```diff
@@ -150,6 +151,8 @@ func uploadContracts(rest *rest.Rest) error {
 			}
 		} else if providerConfig.ConfigType == openzeppelin.OpenzeppelinConfigFile && providerConfig.Solc != nil {
 			configPayload = payloads.ParseOpenZeppelinConfig(providerConfig.Compilers)
+		} else if providerConfig.ConfigType == buidler.BuidlerConfigFile {
+			configPayload = payloads.ParseOpenZeppelinConfig(providerConfig.Compilers)
```
Comment (nebojsa94) [inline commands/push.go:155]: It's better to add a separate function for parsing buidler config file

File: buidler/contract.go [hunk 3]
```diff
@@ -0,0 +1,214 @@
...
+	"io/ioutil"
+	"os"
+	"path/filepath"
+	"runtime"
+	"strconv"
+	"strings"
+)
+
+type BuidlerContract struct {
+	providers.Contract
```
Comment (BogdanHabic) [inline buidler/contract.go:19]: Maybe?
```suggestion
	*providers.Contract
```

File: providers/util.go [hunk 4]
```diff
@@ -22,12 +22,21 @@ func (d DeploymentProviderName) String() string {
...
 
 var letters = []rune("abcdefghijklmnopqrstuvwxyzABCDEFGHIJKLMNOPQRSTUVWXYZ")
 
+var NetworkIdMap = map[string]int{
+	"mainnet": 1,
+	"kovan":   42,
+	"rinkeby": 4,
+	"ropsten": 3,
+	"goerli":  5,
+}
```
Comment (BogdanHabic) [inline providers/util.go:38]: We support other networks as well. Can this be pulled from the API?

**Review by BogdanHabic — COMMENTED** [review]
Comments in comments

**Review by nebojsa94 — APPROVED** [review]

**Pull request merged**